src/storage/SimpleResourceStore.ts

```diff
@@ -56,32 +102,61 @@ export class SimpleResourceStore implements ResourceStore {
     return path;
   }
 
+  /**
+   * Converts the Representation to an array of Quads.
+   * @param representation - Incoming Representation.
+   * @throws {@link UnsupportedMediaTypeHttpError}
+   * If the representation is not a Quad stream.
+   *
+   * @returns Promise of array of Quads pulled from the stream.
+   */
   private async parseRepresentation(representation: Representation): Promise<Quad[]> {
     if (representation.dataType !== 'quad') {
       throw new UnsupportedMediaTypeHttpError('SimpleResourceStore only supports quad representations.');
     }
     return arrayifyStream(representation.data);
   }
 
+  /**
+   * Converts an array of Quads to a Representation.
+   * If preferences.type contains 'text/turtle' the result will be a stream of turtle strings,
+   * otherwise a stream of Quads.
+   * @param data - Quads to transform.
+   * @param preferences - Requested preferences.
+   *
+   * @returns The resulting Representation.
+   */
   private generateRepresentation(data: Quad[], preferences: RepresentationPreferences): Representation {
     if (preferences.type && preferences.type.some((preference): boolean => preference.value.includes('text/turtle'))) {
       return this.generateBinaryRepresentation(data);
     }
     return this.generateQuadRepresentation(data);
   }
 
+  /**
+   * Creates a {@link BinaryRepresentation} of the incoming Quads.
+   * @param data - Quads to transform to text/turtle.
+   *
+   * @returns The resulting binary Representation.
+   */
   private generateBinaryRepresentation(data: Quad[]): BinaryRepresentation {
     return {
       dataType: 'binary',
-      data: streamifyArray(data).pipe(new StreamWriter({ format: 'text/turtle' })),
+      data: streamifyArray([ ...data ]).pipe(new StreamWriter({ format: 'text/turtle' })),
       metadata: { raw: [], profiles: [], contentType: 'text/turtle' },
     };
   }
 
+  /**
+   * Creates a {@link QuadRepresentation} of the incoming Quads.
+   * @param data - Quads to transform to a stream of Quads.
+   *
+   * @returns The resulting quad Representation.
+   */
   private generateQuadRepresentation(data: Quad[]): QuadRepresentation {
     return {
       dataType: 'quad',
-      data: streamifyArray(data),
+      data: streamifyArray([ ...data ]),
```

Comment:
Why needed? (Maybe `streamifyArray` should be updated?)

Reply:
Every time you read an element from the stream, `streamifyArray` pops it from the array.

Reply:
That's weird. Maybe we should just fix `streamifyArray`, but no strong opinion.

Reply:
I think it's to be consistent with `arrayifyStream`, which drains the stream to make an array (since it's impossible to avoid there).